packager/packager.cc

```diff
@@ -23,6 +23,27 @@ namespace {
 constexpr string_view PACKAGE_FILE_NAME = "__package.rb"sv;
 constexpr core::NameRef TEST_NAME = core::Names::Constants::Test();
 
+bool isPackageModule(const core::GlobalState &gs, core::SymbolRef modName) {
+    while (modName.exists() && modName != core::Symbols::root()) {
+        if (modName == core::Symbols::PackageRegistry() || modName == core::Symbols::PackageTests()) {
+            return true;
+        }
+        modName = modName.data(gs)->owner;
+    }
+    return false;
+}
+
+class PrunePackageModules final {
+public:
+    ast::ExpressionPtr postTransformClassDef(core::Context ctx, ast::ExpressionPtr tree) {
+        auto &klass = ast::cast_tree_nonnull<ast::ClassDef>(tree);
+        if (isPackageModule(ctx, klass.symbol)) {
+            return ast::MK::EmptyTree();
```

Comment:
In non-LSP mode it'd be faster if I just leaked these (a decent amount parallelized time is spent de-allocating), but not sure if there's a reasonable way to do that.

Reply:
Nah, don't leak memory. The only way to make deallocation faster is to make smaller trees!

Reply:
I take it back. I understand your issue here. On master there is now a 'leak trees' boolean passed to typecheck!